validator/src/commands/run/execute.rs

```diff
@@ -1409,8 +1398,8 @@ fn get_cluster_shred_version(entrypoints: &[SocketAddr], bind_address: IpAddr) -
     };
     for entrypoint in entrypoints {
         match solana_net_utils::get_cluster_shred_version_with_binding(entrypoint, bind_address) {
-            Err(err) => eprintln!("get_cluster_shred_version failed: {entrypoint}, {err}"),
-            Ok(0) => eprintln!("entrypoint {entrypoint} returned shred-version zero"),
```

Comment:
I think the reason for `eprintln!` instead of `error!` is that the logging isn't initialized yet. so `error!` will not print anything yet until `solana_logger::setup()` is first called.

Reply:
maybe this is a new change.
old code was initializing logging after all those `eprintln`:
https://github.com/anza-xyz/agave/blob/6130ed3acf3581fcd8cf9d537db25ef14c9d5b42/validator/src/main.rs#L932

Reply:
yes to use error! logging must be initialized earlier so i have pushed it up. @steviez is working on a more "permanent" solution with proper error propagation but in the meantime i guess we can have somewhat nicer errors too

Reply:
Not sure how long ago you're talking about, but at least in v2.0, the ordering is same as latest:
- https://github.com/anza-xyz/agave/blob/v2.0/validator/src/main.rs#L920 (setup logging)
- https://github.com/anza-xyz/agave/blob/v2.0/validator/src/main.rs#L1128-L1130 (call `get_cluster_shred_version()`)

Reply:
> but in the meantime i guess we can have somewhat nicer errors too

Luckily, https://github.com/anza-xyz/agave/pull/5026 will be ready for review after CI completes